appsec/src/extension/tags.c

```diff
@@ -1138,6 +1140,60 @@ static PHP_FUNCTION(datadog_appsec_track_custom_event)
     dd_tags_set_sampling_priority();
 }
 
+static PHP_FUNCTION(datadog_appsec_push_params)
+{
+    UNUSED(return_value);
+    if (DDAPPSEC_G(enabled) != ENABLED) {
+        mlog(dd_log_debug, "Trying to access to push_params "
+                           "function while appsec is disabled");
+        return;
+    }
+
+    zval *parameters = NULL;
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "z", &parameters) == FAILURE ||
+        Z_TYPE_P(parameters) != IS_ARRAY) {
+        mlog(dd_log_warning, "Unexpected parameters type. Expected array");
+        return;
+    }
```

Comment:
I did this because we are relying on some functions coming from the frameworks. In case these functions from the frameworks change their return on new versions, this approach will avoid getting php errors.

Reply:
what happens if it isn't an array on the helper side?

Reply:
`path_params` is an array so we should check here. I think the best approach is the one used here. If I get the paramater as a `HashTable`, I would have to tranform it manually to `zval` to add it to `parameters_zv` so PHP core will do that for us.